sdk/src/androidTest/java/com/mongodb/stitch/StitchClientTest.kt

```diff
@@ -0,0 +1,218 @@
+        )).toString()
+
+        /** Mock data for user profile from [StitchClient.getUserProfile] */
+        val mockUserData = JSONObject(mapOf(
+                "domainId" to FakeDomainId,
+                "userId" to FakeUserId,
+                "identities" to arrayOf(mapOf(
+                        "id" to FakeUserId,
+                        "provider" to FakeAnonIdentity)
+                ),
+                "data" to null,
+                "type" to FakeUserType
+        )).toString()
+    }
+
+    /** Base context from test runner */
+    private val instrumentationCtx: Context = InstrumentationRegistry.getContext()
+    /** [StitchClient] for this test */
+    private var stitchClient: StitchClient? = null
+
+    @Before
+    fun setup() {
+        // start the mock server
+        RESTMockServerStarter.startSync(AndroidAssetsFileParser(instrumentationCtx), AndroidLogger())
+
+        // instantiate a new StitchClient using a dummy name and the mock baseUrl
+        val stitchClient = StitchClient(instrumentationCtx, "dummy-app", RESTMockServer.getUrl())
+        this.stitchClient = stitchClient
+
+        // clear all instances of the internal [SharedPreferences] to start with a clean slate
+        stitchClient.properties.clear()
+        stitchClient.javaClass.kotlin.declaredMemberProperties.first {
+            it.name == "_preferences"
+        }.also { it.isAccessible = true }.get(stitchClient).also {
+            (it as SharedPreferences).edit().clear().commit()
+        }
+
+        // mock out all calls related to auth
+        RESTMockServer.whenPOST(pathContains("auth")).thenReturnString(mockAuthData)
+        RESTMockServer.whenGET(pathEndsWith("auth")).thenReturnString(mockProviderData)
+        RESTMockServer.whenDELETE(pathEndsWith("auth")).thenReturnEmpty(200)
+
+        // mock out all calls related to pipelines
+        RESTMockServer.whenPOST(pathEndsWith("pipeline")).thenReturnString(mockPipelineData)
+
+        // mock out user profile call
+        RESTMockServer.whenGET(pathEndsWith("me")).thenReturnString(mockUserData)
+    }
+
+    /**
+     * Test authentication flow
+     */
+    @Test
+    fun testAuth() {
+        var loggedIn = false
+
+        // add an auth listener to the stitchClient, asserting logged in
+        // and logged out status later in this test
+        stitchClient!!.addAuthListener(object : AuthListener {
+            override fun onLogin() {
+                loggedIn = true
+            }
+
+            override fun onLogout(lastProvider: String?) {
+                loggedIn = false
+            }
+        })
+
+        // assert that we have not authenticated yet
+        assertThat(!stitchClient!!.isAuthenticated)
+
+        // fetch mocked authProviders
+        await(stitchClient!!.authProviders).let {
+            // assert all providers provided in mock are available
+            assertThat(it.hasAnonymous() && it.hasEmailPassword()
```

Comment:
Should also probably make sure the `StitchClient`is catching error responses (for mock errors) while we're at it. Will add.